### PR TITLE
Add io-threads-do-reads config to deprecated config table to have no effect.

### LINKED ISSUE #1116: Make io-threads-do-reads harmless
This config is undocumented since #758. The default was changed to "yes" and it is quite useless to set it to "no". Yet, it can happen that some user has an old config file where it is explicitly set to "no". The result will be bad performace, since I/O threads will not do all the I/O.

----

It's indeed confusing.

1. Either remove the whole option from the code. And thus no need for documentation. _OR:_
2. Introduce the option back in the configuration,  just as a comment is fine. And showing the default value "yes": `# io-threads-do-reads yes` with additional text.

_Originally posted by @melroy89 in https://github.com/valkey-io/valkey/discussions/1019#discussioncomment-10824778_

### PR DESCRIPTION
**Pull request by Shivshankar-Reddy**
this fixes: https://github.com/valkey-io/valkey/issues/1116

_Issue details from #1116 by @zuiderkwast_ 

> This config is undocumented since #758. The default was changed to "yes" and it is quite useless to set it to "no". Yet, it can happen that some user has an old config file where it is explicitly set to "no". The result will be bad performace, since I/O threads will not do all the I/O.
> 
> It's indeed confusing.
> 
> 1. Either remove the whole option from the code. And thus no need for documentation. _OR:_
> 2. Introduce the option back in the configuration,  just as a comment is fine. And showing the default value "yes": `# io-threads-do-reads yes` with additional text.
> 
> _Originally posted by @melroy89 in [#1019 (reply in thread)](https://github.com/orgs/valkey-io/discussions/1019#discussioncomment-10824778)_

### CODE REVIEW

**Review by melroy89 — COMMENTED** [review]
Again please update documentation and example config file explaining this option is deprecated and doesn't do anything anymore.

**Comment by Shivshankar-Reddy** [comment]
Added a note in the valkey.conf file as it is deprecated and avoid using it in future. Thanks for remainding it.

**Comment by melroy89** [comment]
Small typo in title: depricated -> deprecated

**Review by zuiderkwast — APPROVED** [review]

File: valkey.conf [hunk 1]
```diff
@@ -1354,7 +1354,11 @@ lazyfree-lazy-user-flush yes
 #
 # prefetch-batch-max-size 16
 #
-# NOTE: If you want to test the server speedup using valkey-benchmark, make
+# NOTE:
+# 1. 'io-threads-do-reads' config is deprecated and will be removed in the future,
+# please avoid using this option if possible.
```
Comment (zuiderkwast) [inline valkey.conf]: Mention also that it has no effect? (Some deprecated features still work, but this one doesn't.)

```suggestion
# 1. The 'io-threads-do-reads' config is deprecated and has no effect. It 
will be removed in the future. Please avoid using this option if possible.
```

Comment (madolson) [reply valkey.conf]: We won't necessarily remove it in the feature, we shouldn't use such language.

Comment (zuiderkwast) [reply valkey.conf]: Sorry for merging too fast. Shall we just delete the sentence "will be removed in the future"? A follow-up PR is welcome.

Comment (Shivshankar-Reddy) [reply valkey.conf]: Sure @madolson, I have noted your point and created https://github.com/valkey-io/valkey/pull/1150 to revoke the statement from valkey.conf.

**Pull request merged**